### PR TITLE
PROD-199: for immediate origination return a kapps_call if successful

### PR DESCRIPTION
**Pull request by k-anderson**
Tweaks to handle immediate origination requests in addition to adding the kapps_call on the reply if successful so that the requestor doesnt need to attempt to construct it, rather can just start immediately issuing call commands.

### CODE REVIEW

**Comment by k-anderson** [comment]
Related PRs:
https://github.com/2600hz/kazoo-core/pull/274
https://github.com/2600hz/kazoo-stepswitch/pull/18
https://github.com/2600hz/kazoo-ecallmgr/pull/94
https://github.com/2600hz/kazoo-crossbar/pull/193

File: src/stepswitch_originate.erl [hunk 1]
```diff
@@ -234,6 +236,8 @@ build_originate(#state{endpoints=Endpoints
                    ,{<<"From-URI">>, originate_from_uri(CIDNum, OffnetReq)}
                    ,{<<"Realm">>, stepswitch_util:default_realm(OffnetReq)}
                    ,{<<"Reseller-ID">>, kz_services_reseller:get_id(AccountId)}
+                   ,{<<"Authorizing-ID">>, AccountId}
+                   ,{<<"Authorizing-Type">>, <<"account">>}
```
Comment (lazedo) [inline src/stepswitch_originate.erl:240]: why this change here ? i think this could break b-leg authorizing-id for cfwd

**Review by lazedo — COMMENTED** [review]
need clarification

File: src/stepswitch_originate.erl (hunk 1 above)
Comment (k-anderson) [reply src/stepswitch_originate.erl:240]: I too have a little bit of concern about this... but call forward doesn't use originate...right? ;)

forward -> loopback -> hunt -> callflow no_match -> stepswitch resources -> bridge

I will look back at the code but I think only faxes originate, because this wouldn't be a b-leg.  It creates a one-sided a-leg... (that might later be intercepted but isn't by default part of a bridge/existing call)... what about if I make it add if not present already?

Comment (lazedo) [reply src/stepswitch_originate.erl:240]: thanks for the insight, you are right, `cfwd` should not be affected

Comment (lazedo) [reply src/stepswitch_originate.erl:240]: need @mark2600 to validate this as `qubicle` does use originate ?

**Review by lazedo — APPROVED** [review]

File: src/stepswitch_originate.erl [hunk 2]
```diff
@@ -320,11 +324,14 @@ originate_error(JObj, OffnetReq) ->
 -spec originate_success(kz_json:object(), kz_json:object()) -> kz_term:proplist().
 originate_success(JObj, OffnetReq) ->
     lager:debug("originate request successfully completed"),
-    [{<<"Call-ID">>, kz_json:get_first_defined([<<"Origination-Call-ID">>, <<"Outbound-Call-ID">>], OffnetReq)}
-    ,{<<"Msg-ID">>, kz_api:msg_id(OffnetReq)}
+    ResourceResponse =
+        kz_json:set_value(<<"Call-ID">>, kz_json:get_value(<<"Call-ID">>, JObj), kz_api:remove_defaults(JObj)),
```
Comment (lazedo) [inline src/stepswitch_originate.erl:328]:
```suggestion
    ResourceResponse = kz_api:remove_defaults(JObj),
```
setting call-id with value from call-id

File: src/stepswitch_originate.erl [hunk 3]
```diff
@@ -320,11 +324,14 @@ originate_error(JObj, OffnetReq) ->
...
+    [{<<"Msg-ID">>, kz_api:msg_id(OffnetReq)}
+    ,{<<"Call-ID">>, kz_json:get_first_defined([<<"Origination-Call-ID">>, <<"Outbound-Call-ID">>], OffnetReq)}
+    ,{<<"Call">>, kapps_call:to_json(kapps_call:from_originate_resp(JObj))}
+    ,{<<"Resource-Response">>, ResourceResponse}
```
Comment (lazedo) [inline src/stepswitch_originate.erl:332]: we are duplicating the payload here. how about using kapps_call:from_originate_resp/1 in the caller from `Resource-Response`  field ?

File: src/stepswitch_originate.erl (hunk 2 above)
Comment (lazedo) [reply src/stepswitch_originate.erl:328]: disregard this, `kz_api:remove_defaults` removes the Call-ID
